frontend/src/core/comments/components/AddComment.js

```diff
@@ -1,97 +1,393 @@
+        const commentEditorSpan = document.querySelector('.comments-list .add-comment .comment-editor p span');
+        const commentEditorSpanHeight = !commentEditorSpan ? 0 : commentEditorSpan.offsetHeight;
+        const setTopAdjustment = commentEditorBottomPadding + commentEditorSpanHeight;
+        const suggestionsHeightAdjustment = (
+            commentEditorTopPadding + ((commentEditorLineHeight - commentEditorSpanHeight) / 2));
+
+        let setTop = (rect.top + window.pageYOffset) + setTopAdjustment;
+        let setLeft = rect.left + window.pageXOffset;
+
+        // If suggestions overflow the window or teams container height then adjust the
+        // position so they display above the comment
+        const suggestionsHeight = el.clientHeight + suggestionsHeightAdjustment;
+        const teamCommentsOverflow = !teamCommentsRect ? false : 
+            ((setTop + el.clientHeight) - tabIndexHeight) > teamCommentsRect.height;
+
+        if ((teamCommentsActive && teamCommentsOverflow) || setTop + suggestionsHeight > window.innerHeight) { 
+            setTop = ((rect.top + window.pageYOffset) - suggestionsHeight);
+        }
 
-    const handleOnChange = () => {
-        const textAreaLineHeight = 24;
-        commentInput.current.rows = minRows;
+        // If suggestions in team comments scroll below or suggestions in translation
+        // comments scroll above the next section or overflow the window then hide the suggestions
+        if ((teamCommentsRect && teamCommentsActive && 
+                (((setTop + suggestionsHeight) - editorMenuHeight) > teamCommentsRect.height)) ||
+            (translateCommentsRect && translateCommentsActive && (rect.top < translateCommentsRect.top)) || 
+            (translateCommentsRect && translateCommentsActive && (setTop + suggestionsHeight > window.innerHeight))) {
+            el.style.display = 'none';
+        }
+        
+        // If suggestions overflow the window width in team comments or the right side of the 
+        // translations comments then adjust the position so they display to the left of the mention
+        const suggestionsWidth = el.clientWidth
+        const translateCommentsOverflow = !translateCommentsRect ? false :
+            setLeft + suggestionsWidth > translateCommentsRect.right;
+        
+        if (setLeft + suggestionsWidth > window.innerWidth || 
+            (translateCommentsActive && translateCommentsOverflow)) {
+            setLeft = rect.right - suggestionsWidth;
+        }
 
-        const currentRows = Math.trunc(commentInput.current.scrollHeight / textAreaLineHeight);
+        el.style.top = `${ setTop }px`;
+        el.style.left = `${ setLeft }px`;
+    
+    }, [ chars.length, editor, index, search, target, scrollPosition ]);
 
-        if (currentRows < maxRows) {
-            commentInput.current.rows = currentRows;
+    React.useEffect(() => {
```

Comment:
I think effects should all have a description comment, just like we would do for functions. In a component like this one, where the complexity is very high, it's quite hard to navigate all the hooks without getting lost. Can you please add a description of what each complex hook does in this file?